e2e/tests/keycloak-openmrs-flows.spec.ts

```diff
@@ -1,191 +1,183 @@
 import { test, expect } from '@playwright/test';
-import { HomePage, delay, randomKeycloakRoleName } from '../utils/functions/testBase';
-import { randomOpenMRSRoleName, randomSupersetRoleName } from '../utils/functions/testBase';
+import { Keycloack, randomKeycloakRoleName } from '../utils/functions/keycloak';
+import { OpenMRS, delay, randomOpenMRSRoleName } from '../utils/functions/openmrs';
+import { Superset, randomSupersetRoleName} from '../utils/functions/superset';
 import { O3_URL, KEYCLOAK_URL } from '../utils/configs/globalSetup';
 
-let homePage: HomePage;
+let openmrs: OpenMRS;
+let keycloack: Keycloack;
+let superset: Superset;
 
 test.beforeEach(async ({ page }) => {
-  homePage = new HomePage(page);
+  openmrs = new OpenMRS(page);
+  keycloack = new Keycloack(page);
+  superset = new Superset(page);
+
+  await openmrs.initiateLogin();
+  await expect(page).toHaveURL(/.*home/);
 });
 
 test('Creating an OpenMRS role creates the corresponding Keycloak role.', async ({ page }) => {
   // replay
-  await homePage.initiateLogin();
-  await expect(page).toHaveURL(/.*home/);
   await page.goto(`${O3_URL}/openmrs/admin/users/role.list`);
-  await homePage.addOpenMRSRole();
+  await openmrs.addOpenMRSRole();
```

Comment:
Since the context is known ("openmrs"), there is no need to be specific in the method name anymore:

`openmrs.addOpenMRSRole()` → `openmrs.addRole()`

And then the same logic would apply everywhere else of course, for all components' pages.